airflow/api_fastapi/core_api/routes/public/assets.py

```diff
@@ -134,6 +138,36 @@ def get_asset_events(
     )
 
 
+@assets_router.post(
+    "/events",
+    responses=create_openapi_http_exception_doc([401, 403, 404]),
+)
+def create_asset_events(
+    create_asset_request: CreateAssetEventsBody,
+    session: Annotated[Session, Depends(get_session)],
+) -> AssetEventResponse:
+    """Create asset events."""
+    asset = session.scalar(
+        select(AssetModel).where(AssetModel.uri == create_asset_request.asset_uri).limit(1)
+    )
+    if not asset:
+        raise HTTPException(404, f"Asset with uri: `{create_asset_request.asset_uri}` was not found")
+    timestamp = timezone.utcnow()
+
+    create_asset_request.extra["from_rest_api"] = True
```

Comment:
This can be moved into a field validator directly in the Datamodels. (removing the need to do that manually here).

Reply:
done